dashboard/signals.py

```diff
@@ -101,6 +75,8 @@ def handle_delete_certificate(sender, instance, **kwargs):  # pylint: disable=un
     """
     user = instance.user
     program = instance.course_run.course.program
-    program_enrollment = ProgramEnrollment.objects.filter(user=user, program=program).first()
-    if program_enrollment is not None:
+    try:
+        program_enrollment = ProgramEnrollment.objects.get(user=user, program=program)
         index_program_enrolled_users.delay([program_enrollment])
+    except ProgramEnrollment.DoesNotExist:
+        pass
```

Comment:
this try/except block appears 4 times. can you create a helper method (`index_enrollment_if_exists` et al) that takes a user and program and runs these lines?